File: web-login-core/src/main/java/com/example/weblogincore/domain/model/form/Question.java
package com.example.weblogincore.domain.model.form;

public abstract class Question {
    private final Long id;
    private final String title;
    private final String description;
    private final String imageUrl;
    private final Boolean isMandatory;

    public Question(Long id, String title, String description, String imageUrl, Boolean isMandatory) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.imageUrl = imageUrl;
        this.isMandatory = isMandatory;
    }

    public abstract Integer totalQuestions();

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public Boolean getMandatory() {
        return isMandatory;
    }
}

File: web_login_flutter/lib/models/form/question.dart
import 'package:equatable/equatable.dart';

abstract class Question extends Equatable{
  final int _id;
  final String _title;
  final String? _description;
  final String? _imageUrl;
  final bool _isMandatory;

  const Question({required id, required title, required description, required imageUrl, required isMandatory}) :
      _id = id, _title = title, _description = description, _imageUrl = imageUrl, _isMandatory = isMandatory;

  @override
  String toString() {
    return 'Question{_id: $_id, _title: $_title, _description: $_description, _imageUrl: $_imageUrl, _isMandatory: $_isMandatory}';
  }

  int get id => _id;

  bool get isMandatory => _isMandatory;

  String? get imageUrl => _imageUrl;

  String? get description => _description;

  String get title => _title;
}